// react-audio-voice-recorder/dist/react-audio-visualize.es.0ccda538.mjs
import Ce, { useState as Y, useRef as $e, useEffect as H, useCallback as fr } from "react";
var ne = { exports: {} }, W = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Te;
function pr() {
  if (Te)
    return W;
  Te = 1;
  var p = Ce, y = Symbol.for("react.element"), _ = Symbol.for("react.fragment"), b = Object.prototype.hasOwnProperty, g = p.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, w = { key: !0, ref: !0, __self: !0, __source: !0 };
  function k(i, c, h) {
    var o, d = {}, l = null, O = null;
    h !== void 0 && (l = "" + h), c.key !== void 0 && (l = "" + c.key), c.ref !== void 0 && (O = c.ref);
    for (o in c)
      b.call(c, o) && !w.hasOwnProperty(o) && (d[o] = c[o]);
    if (i && i.defaultProps)
      for (o in c = i.defaultProps, c)
        d[o] === void 0 && (d[o] = c[o]);
    return { $$typeof: y, type: i, key: l, ref: O, props: d, _owner: g.current };
  }
  return W.Fragment = _, W.jsx = k, W.jsxs = k, W;
}
var q = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var De;
function yr() {
  return De || (De = 1, process.env.NODE_ENV !== "production" && function() {
    var p = Ce, y = Symbol.for("react.element"), _ = Symbol.for("react.portal"), b = Symbol.for("react.fragment"), g = Symbol.for("react.strict_mode"), w = Symbol.for("react.profiler"), k = Symbol.for("react.provider"), i = Symbol.for("react.context"), c = Symbol.for("react.forward_ref"), h = Symbol.for("react.suspense"), o = Symbol.for("react.suspense_list"), d = Symbol.for("react.memo"), l = Symbol.for("react.lazy"), O = Symbol.for("react.offscreen"), E = Symbol.iterator, $ = "@@iterator";
    function T(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = E && e[E] || e[$];
      return typeof r == "function" ? r : null;
    }
    var u = p.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function m(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), n = 1; n < r; n++)
          t[n - 1] = arguments[n];
        D("error", e, t);
      }
    }
    function D(e, r, t) {
      {
        var n = u.ReactDebugCurrentFrame, f = n.getStackAddendum();
        f !== "" && (r += "%s", t = t.concat([f]));
        var v = t.map(function(s) {
          return String(s);
        });
        v.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, v);
      }
    }
    var Fe = !1, Ne = !1, Pe = !1, Ae = !1, Ie = !1, oe;
    oe = Symbol.for("react.module.reference");
    function Le(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === b || e === w || Ie || e === g || e === h || e === o || Ae || e === O || Fe || Ne || Pe || typeof e == "object" && e !== null && (e.$$typeof === l || e.$$typeof === d || e.$$typeof === k || e.$$typeof === i || e.$$typeof === c || e.$$typeof === oe || e.getModuleId !== void 0));
    }
    function We(e, r, t) {
      var n = e.displayName;
      if (n)
        return n;
      var f = r.displayName || r.name || "";
      return f !== "" ? t + "(" + f + ")" : t;
    }
    function ae(e) {
      return e.displayName || "Context";
    }
    function x(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && m("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case b:
          return "Fragment";
        case _:
          return "Portal";
        case w:
          return "Profiler";
        case g:
          return "StrictMode";
        case h:
          return "Suspense";
        case o:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case i:
            var r = e;
            return ae(r) + ".Consumer";
          case k:
            var t = e;
            return ae(t._context) + ".Provider";
          case c:
            return We(e, e.render, "ForwardRef");
          case d:
            var n = e.displayName || null;
            return n !== null ? n : x(e.type) || "Memo";
          case l: {
            var f = e, v = f._payload, s = f._init;
            try {
              return x(s(v));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var N = Object.assign, I = 0, ie, ce, se, le, ue, fe, pe;
    function ye() {
    }
    ye.__reactDisabledLog = !0;
    function ze() {
      {
        if (I === 0) {
          ie = console.log, ce = console.info, se = console.warn, le = console.error, ue = console.group, fe = console.groupCollapsed, pe = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: ye,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        I++;
      }
    }
    function Me() {
      {
        if (I--, I === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: N({}, e, {
              value: ie
            }),
            info: N({}, e, {
              value: ce
            }),
            warn: N({}, e, {
              value: se
            }),
            error: N({}, e, {
              value: le
            }),
            group: N({}, e, {
              value: ue
            }),
            groupCollapsed: N({}, e, {
              value: fe
            }),
            groupEnd: N({}, e, {
              value: pe
            })
          });
        }
        I < 0 && m("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var X = u.ReactCurrentDispatcher, J;
    function z(e, r, t) {
      {
        if (J === void 0)
          try {
            throw Error();
          } catch (f) {
            var n = f.stack.trim().match(/\n( *(at )?)/);
            J = n && n[1] || "";
          }
        return `
` + J + e;
      }
    }
    var K = !1, M;
    {
      var Ue = typeof WeakMap == "function" ? WeakMap : Map;
      M = new Ue();
    }
    function de(e, r) {
      if (!e || K)
        return "";
      {
        var t = M.get(e);
        if (t !== void 0)
          return t;
      }
      var n;
      K = !0;
      var f = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var v;
      v = X.current, X.current = null, ze();
      try {
        if (r) {
          var s = function() {
            throw Error();
          };
          if (Object.defineProperty(s.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(s, []);
            } catch (F) {
              n = F;
            }
            Reflect.construct(e, [], s);
          } else {
            try {
              s.call();
            } catch (F) {
              n = F;
            }
            e.call(s.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (F) {
            n = F;
          }
          e();
        }
      } catch (F) {
        if (F && n && typeof F.stack == "string") {
          for (var a = F.stack.split(`
`), j = n.stack.split(`
`), S = a.length - 1, R = j.length - 1; S >= 1 && R >= 0 && a[S] !== j[R]; )
            R--;
          for (; S >= 1 && R >= 0; S--, R--)
            if (a[S] !== j[R]) {
              if (S !== 1 || R !== 1)
                do
                  if (S--, R--, R < 0 || a[S] !== j[R]) {
                    var C = `
` + a[S].replace(" at new ", " at ");
                    return e.displayName && C.includes("<anonymous>") && (C = C.replace("<anonymous>", e.displayName)), typeof e == "function" && M.set(e, C), C;
                  }
                while (S >= 1 && R >= 0);
              break;
            }
        }
      } finally {
        K = !1, X.current = v, Me(), Error.prepareStackTrace = f;
      }
      var A = e ? e.displayName || e.name : "", Ee = A ? z(A) : "";
      return typeof e == "function" && M.set(e, Ee), Ee;
    }
    function Be(e, r, t) {
      return de(e, !1);
    }
    function Ve(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function U(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return de(e, Ve(e));
      if (typeof e == "string")
        return z(e);
      switch (e) {
        case h:
          return z("Suspense");
        case o:
          return z("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case c:
            return Be(e.render);
          case d:
            return U(e.type, r, t);
          case l: {
            var n = e, f = n._payload, v = n._init;
            try {
              return U(v(f), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var B = Object.prototype.hasOwnProperty, me = {}, he = u.ReactDebugCurrentFrame;
    function V(e) {
      if (e) {
        var r = e._owner, t = U(e.type, e._source, r ? r.type : null);
        he.setExtraStackFrame(t);
      } else
        he.setExtraStackFrame(null);
    }
    function qe(e, r, t, n, f) {
      {
        var v = Function.call.bind(B);
        for (var s in e)
          if (v(e, s)) {
            var a = void 0;
            try {
              if (typeof e[s] != "function") {
                var j = Error((n || "React class") + ": " + t + " type `" + s + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[s] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw j.name = "Invariant Violation", j;
              }
              a = e[s](r, s, n, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (S) {
              a = S;
            }
            a && !(a instanceof Error) && (V(f), m("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", n || "React class", t, s, typeof a), V(null)), a instanceof Error && !(a.message in me) && (me[a.message] = !0, V(f), m("Failed %s type: %s", t, a.message), V(null));
          }
      }
    }
    var Ye = Array.isArray;
    function Q(e) {
      return Ye(e);
    }
    function He(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function Xe(e) {
      try {
        return ve(e), !1;
      } catch {
        return !0;
      }
    }
    function ve(e) {
      return "" + e;
    }
    function ge(e) {
      if (Xe(e))
        return m("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", He(e)), ve(e);
    }
    var L = u.ReactCurrentOwner, Je = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, be, we, Z;
    Z = {};
    function Ke(e) {
      if (B.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function Qe(e) {
      if (B.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function Ze(e, r) {
      if (typeof e.ref == "string" && L.current && r && L.current.stateNode !== r) {
        var t = x(L.current.type);
        Z[t] || (m('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', x(L.current.type), e.ref), Z[t] = !0);
      }
    }
    function Ge(e, r) {
      {
        var t = function() {
          be || (be = !0, m("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function er(e, r) {
      {
        var t = function() {
          we || (we = !0, m("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var rr = function(e, r, t, n, f, v, s) {
      var a = {
        $$typeof: y,
        type: e,
        key: r,
        ref: t,
        props: s,
        _owner: v
      };
      return a._store = {}, Object.defineProperty(a._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(a, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: n
      }), Object.defineProperty(a, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: f
      }), Object.freeze && (Object.freeze(a.props), Object.freeze(a)), a;
    };
    function tr(e, r, t, n, f) {
      {
        var v, s = {}, a = null, j = null;
        t !== void 0 && (ge(t), a = "" + t), Qe(r) && (ge(r.key), a = "" + r.key), Ke(r) && (j = r.ref, Ze(r, f));
        for (v in r)
          B.call(r, v) && !Je.hasOwnProperty(v) && (s[v] = r[v]);
        if (e && e.defaultProps) {
          var S = e.defaultProps;
          for (v in S)
            s[v] === void 0 && (s[v] = S[v]);
        }
        if (a || j) {
          var R = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          a && Ge(s, R), j && er(s, R);
        }
        return rr(e, a, j, f, n, L.current, s);
      }
    }
    var G = u.ReactCurrentOwner, Se = u.ReactDebugCurrentFrame;
    function P(e) {
      if (e) {
        var r = e._owner, t = U(e.type, e._source, r ? r.type : null);
        Se.setExtraStackFrame(t);
      } else
        Se.setExtraStackFrame(null);
    }
    var ee;
    ee = !1;
    function re(e) {
      return typeof e == "object" && e !== null && e.$$typeof === y;
    }
    function ke() {
      {
        if (G.current) {
          var e = x(G.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function nr(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var Re = {};
    function or(e) {
      {
        var r = ke();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function _e(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = or(r);
        if (Re[t])
          return;
        Re[t] = !0;
        var n = "";
        e && e._owner && e._owner !== G.current && (n = " It was passed a child from " + x(e._owner.type) + "."), P(e), m('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, n), P(null);
      }
    }
    function Oe(e, r) {
      {
        if (typeof e != "object")
          return;
        if (Q(e))
          for (var t = 0; t < e.length; t++) {
            var n = e[t];
            re(n) && _e(n, r);
          }
        else if (re(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var f = T(e);
          if (typeof f == "function" && f !== e.entries)
            for (var v = f.call(e), s; !(s = v.next()).done; )
              re(s.value) && _e(s.value, r);
        }
      }
    }
    function ar(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === c || r.$$typeof === d))
          t = r.propTypes;
        else
          return;
        if (t) {
          var n = x(r);
          qe(t, e.props, "prop", n, e);
        } else if (r.PropTypes !== void 0 && !ee) {
          ee = !0;
          var f = x(r);
          m("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", f || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && m("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function ir(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var n = r[t];
          if (n !== "children" && n !== "key") {
            P(e), m("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", n), P(null);
            break;
          }
        }
        e.ref !== null && (P(e), m("Invalid attribute `ref` supplied to `React.Fragment`."), P(null));
      }
    }
    function je(e, r, t, n, f, v) {
      {
        var s = Le(e);
        if (!s) {
          var a = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (a += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var j = nr(f);
          j ? a += j : a += ke();
          var S;
          e === null ? S = "null" : Q(e) ? S = "array" : e !== void 0 && e.$$typeof === y ? (S = "<" + (x(e.type) || "Unknown") + " />", a = " Did you accidentally export a JSX literal instead of a component?") : S = typeof e, m("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", S, a);
        }
        var R = tr(e, r, t, f, v);
        if (R == null)
          return R;
        if (s) {
          var C = r.children;
          if (C !== void 0)
            if (n)
              if (Q(C)) {
                for (var A = 0; A < C.length; A++)
                  Oe(C[A], e);
                Object.freeze && Object.freeze(C);
              } else
                m("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Oe(C, e);
        }
        return e === b ? ir(R) : ar(R), R;
      }
    }
    function cr(e, r, t) {
      return je(e, r, t, !0);
    }
    function sr(e, r, t) {
      return je(e, r, t, !1);
    }
    var lr = sr, ur = cr;
    q.Fragment = b, q.jsx = lr, q.jsxs = ur;
  }()), q;
}
process.env.NODE_ENV === "production" ? ne.exports = pr() : ne.exports = yr();
var xe = ne.exports;
const dr = (p, y, _, b) => {
  let g = y / (_ + b), w = Math.floor(p.length / g);
  g > p.length && (g = p.length, w = 1);
  const k = [];
  for (let i = 0; i < g; i++) {
    let c = 0;
    for (let h = 0; h < w && i * w + h < p.length; h++)
      c += p[i * w + h];
    k.push(c / w);
  }
  return k;
}, mr = (p, y, _, b, g, w) => {
  const k = y.height / 2, i = y.getContext("2d");
  i && (i.clearRect(0, 0, y.width, y.height), g !== "transparent" && (i.fillStyle = g, i.fillRect(0, 0, y.width, y.height)), p.forEach((c, h) => {
    i.fillStyle = w;
    const o = h * (_ + b), d = k - c / 2, l = _, O = c || 1;
    i.beginPath(), i.roundRect(o, d, l, O, 50), i.fill();
  }));
}, gr = ({
  mediaRecorder: p,
  width: y = "100%",
  height: _ = "100%",
  barWidth: b = 2,
  gap: g = 1,
  backgroundColor: w = "transparent",
  barColor: k = "rgb(160, 198, 255)",
  fftSize: i = 1024,
  maxDecibels: c = -10,
  minDecibels: h = -90,
  smoothingTimeConstant: o = 0.4
}) => {
  const [d] = Y(() => new AudioContext()), [l, O] = Y(), E = $e(null);
  H(() => {
    if (!p.stream)
      return;
    const u = d.createAnalyser();
    O(u), u.fftSize = i, u.minDecibels = h, u.maxDecibels = c, u.smoothingTimeConstant = o, d.createMediaStreamSource(p.stream).connect(u);
  }, [p.stream]), H(() => {
    (l != null ? l : p.state === "recording") && $();
  }, [l, p.state]);
  const $ = fr(() => {
    if (!l)
      return;
    const u = new Uint8Array(l == null ? void 0 : l.frequencyBinCount);
    p.state === "recording" ? (l == null || l.getByteFrequencyData(u), T(u), requestAnimationFrame($)) : p.state === "paused" ? T(u) : p.state === "inactive" && d.state !== "closed" && d.close();
  }, [l]), T = (u) => {
    if (!E.current)
      return;
    const m = dr(
      u,
      E.current.width,
      b,
      g
    );
    mr(
      m,
      E.current,
      b,
      g,
      w,
      k
    );
  };
  return /* @__PURE__ */ xe.jsx(
    "canvas",
    {
      ref: E,
      width: y,
      height: _,
      style: {
        aspectRatio: "unset"
      }
    }
  );
}, hr = (p, y, _, b, g) => {
  const w = p.getChannelData(0), k = _ / (b + g), i = Math.floor(w.length / k), c = y / 2;
  let h = [], o = 0;
  for (let d = 0; d < k; d++) {
    const l = [];
    let O = 0;
    const E = [];
    let $ = 0;
    for (let m = 0; m < i && d * i + m < p.length; m++) {
      const D = w[d * i + m];
      D <= 0 && (l.push(D), O++), D > 0 && (E.push(D), $++);
    }
    const T = l.reduce((m, D) => m + D, 0) / O, u = { max: E.reduce((m, D) => m + D, 0) / $, min: T };
    u.max > o && (o = u.max), Math.abs(u.min) > o && (o = Math.abs(u.min)), h.push(u);
  }
  if (c * 0.8 > o * c) {
    const d = c * 0.8 / o;
    h = h.map((l) => ({
      max: l.max * d,
      min: l.min * d
    }));
  }
  return h;
}, te = (p, y, _, b, g, w, k, i = 0, c = 1) => {
  const h = y.height / 2, o = y.getContext("2d");
  if (!o)
    return;
  o.clearRect(0, 0, y.width, y.height), g !== "transparent" && (o.fillStyle = g, o.fillRect(0, 0, y.width, y.height));
  const d = (i || 0) / c;
  p.forEach((l, O) => {
    const E = O / p.length, $ = d > E;
    o.fillStyle = $ && k ? k : w;
    const T = O * (_ + b), u = h + l.min, m = _, D = h + l.max - u;
    o.beginPath(), o.roundRect(T, u, m, D, 50), o.fill();
  });
}, br = ({
  blob: p,
  width: y,
  height: _,
  barWidth: b = 2,
  gap: g = 1,
  currentTime: w,
  style: k,
  backgroundColor: i = "transparent",
  barColor: c = "rgb(184, 184, 184)",
  barPlayedColor: h = "rgb(160, 198, 255)"
}) => {
  const o = $e(null), [d, l] = Y([]), [O, E] = Y(0);
  return H(() => {
    (async () => {
      if (!o.current)
        return;
      if (!p) {
        const T = Array.from({ length: 100 }, () => ({
          max: 0,
          min: 0
        }));
        te(
          T,
          o.current,
          b,
          g,
          i,
          c,
          h
        );
        return;
      }
      const $ = await p.arrayBuffer();
      await new AudioContext().decodeAudioData($, (T) => {
        if (!o.current)
          return;
        E(T.duration);
        const u = hr(T, _, y, b, g);
        l(u), te(
          u,
          o.current,
          b,
          g,
          i,
          c,
          h
        );
      });
    })();
  }, [p, o.current]), H(() => {
    o.current && te(
      d,
      o.current,
      b,
      g,
      i,
      c,
      h,
      w,
      O
    );
  }, [w, O]), /* @__PURE__ */ xe.jsx(
    "canvas",
    {
      ref: o,
      width: y,
      height: _,
      style: {
        ...k
      }
    }
  );
};
export {
  br as AudioVisualizer,
  gr as LiveAudioVisualizer
};
